improver/developer_tools/metadata_interpreter.py

```diff
@@ -0,0 +1,563 @@
+    PERC = "percentiles"
+    DIAG = "realizations"
+    ANCIL = "ancillary"
+
+    def __init__(self):
+        """Initialise class parameters"""
+        self.model_id_attr = "mosg__model_configuration"
+        self.unhandled = False
+
+        # set up empty strings to record any non-compliance (returned as one error
+        # after all checks have been made) or warnings
+        self.errors = []
+        self.warnings = []
+        # initialise information to be derived from input cube
+        self.prod_type = "gridded"  # gridded or spot
+        self.field_type = (
+            None  # probabilities, percentiles, realizations, ancillary or name
+        )
+        self.diagnostic = None  # name
+        self.relative_to_threshold = None  # for probability data only
+        self.methods = ""  # human-readable interpretation of cell method(s)
+        self.post_processed = (
+            None  # True / False on whether significant processing applied
+        )
+        self.model = None  # human-readable model name
+        self.blended = None  # has it been model blended (True / False)
+
+    def check_probability_cube_metadata(self, cube):
+        """Checks probability-specific metadata"""
+        if cube.units != "1":
+            self.errors.append(
+                f"Expected units of 1 on probability data, got {cube.units}"
+            )
+
+        try:
+            self.diagnostic = get_diagnostic_cube_name_from_probability_name(
+                cube.name()
+            )
+        except ValueError as cause:
+            # if the probability name is not valid
+            self.errors.append(str(cause))
+
+        expected_threshold_name = get_threshold_coord_name_from_probability_name(
+            cube.name()
+        )
+
+        if not cube.coords(expected_threshold_name):
+            msg = f"Cube does not have expected threshold coord '{expected_threshold_name}'; "
+            try:
+                threshold_name = find_threshold_coordinate(cube).name()
+            except CoordinateNotFoundError:
+                coords = [coord.name() for coord in cube.coords()]
+                msg = (
+                    f"no coord with var_name='threshold' found in all coords: {coords}"
+                )
+                self.errors.append(msg)
+            else:
+                msg += f"threshold coord has incorrect name '{threshold_name}'"
+                self.errors.append(msg)
+                self.check_threshold_coordinate_properties(
+                    cube.name(), cube.coord(threshold_name)
+                )
+        else:
+            threshold_coord = cube.coord(expected_threshold_name)
+            self.check_threshold_coordinate_properties(cube.name(), threshold_coord)
+
+    def check_threshold_coordinate_properties(self, cube_name, threshold_coord):
+        """Checks threshold coordinate properties are correct and consistent with
+        cube name"""
+        threshold_var_name = threshold_coord.var_name
```

Comment:
"threshold_var_name" is only used once and it is on the next line. I suggest getting rid of it.

Reply:
OK